### PR TITLE
Fixes #14464: If a relay is deleted (node->delete) a system rule for it remains

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/14464

### CODE REVIEW

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/PolicyServerManagementService.scala [hunk 1]
```diff
@@ -140,6 +153,36 @@ class PolicyServerManagementServiceImpl(
...
+   * - group: nodeGroupId=hasPolicyServer-${uuid},groupCategoryId=SystemGroups,groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration
+   * - rule target:  ruleTarget=policyServer:${RELAY_UUID},groupCategoryId=SystemGroups,groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration
+   * - directive:  directiveId=${RELAY_UUID}-distributePolicy,activeTechniqueId=distributePolicy,techniqueCategoryId=Rudder Internal,techniqueCategoryId=Active Techniques,ou=Rudder,cn=rudder-configuration
+   * - directive: directiveId=common-${RELAY_UUID},activeTechniqueId=common,techniqueCategoryId=Rudder Internal,techniqueCategoryId=Active Techniques,ou=Rudder,cn=rudder-configuratio
+   * - rule: ruleId=${RELAY_UUID}-DP,ou=Rules,ou=Rudder,cn=rudder-configuration
+   * - rule: ruleId=hasPolicyServer-${RELAY_UUID},ou=Rules,ou=Rudder,cn=rudder-configuration
+   */
+  def deleteRelaySystemObjects(policyServerId: NodeId): Box[Unit] = {
+    if(policyServerId == Constants.ROOT_POLICY_SERVER_ID) {
+      Failure("Root server coniguration elements can't be deleted")
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/PolicyServerManagementService.scala]:
```suggestion
      Failure("Root server configuration elements can't be deleted")
```

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/PolicyServerManagementService.scala [hunk 2]
```diff
@@ -140,6 +153,36 @@ class PolicyServerManagementServiceImpl(
...
+      for {
+        con <- ldap
+        _   <- con.delete(DN(s"nodeGroupId=hasPolicyServer-${id}", dit.GROUP.SYSTEM.dn))
+        _   <- con.delete(DN(s"ruleTarget=policyServer:${id}", dit.GROUP.SYSTEM.dn))
+        _   <- con.delete(DN(s"directiveId=${id}-distributePolicy,activeTechniqueId=distributePolicy,techniqueCategoryId=Rudder Internal", dit.ACTIVE_TECHNIQUES_LIB.dn))
+        _   <- con.delete(DN(s"directiveId=common-${id},activeTechniqueId=common,techniqueCategoryId=Rudder Internal", dit.ACTIVE_TECHNIQUES_LIB.dn))
+        _   <- con.delete(DN(s"ruleId=${id}-DP", dit.RULES.dn))
+        _   <- con.delete(DN(s"ruleId=hasPolicyServer-${id}", dit.RULES.dn))
+      } yield ()
+    }
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/PolicyServerManagementService.scala]: there aren't any log there - if something goes bad, we'll never know.
And a debug to know that is went fine would be great

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/RemoveNodeService.scala [hunk 3]
```diff
@@ -202,10 +203,13 @@ class RemoveNodeServiceImpl(
           for {
             moved  <- groupLibMutex.writeLock {atomicDelete(nodeId, modId, actor) } ?~! "Error when deleting a node"
             closed <- nodeConfigurationsRepo.closeNodeConfigurations(nodeId)
-
             invLogDetails = InventoryLogDetails (nodeInfo.id,nodeInfo.inventoryDate, nodeInfo.hostname, nodeInfo.osDetails.fullName, actor.name)
             eventlog = DeleteNodeEventLog.fromInventoryLogDetails (None, actor, invLogDetails)
             saved  <- actionLogger.saveEventLog(modId, eventlog)
+                      // if the node was a policy server, we need to delete group and directive/rule for it
+            _      <- if(nodeInfo.isPolicyServer) {
+                        policyServerManagement.deleteRelaySystemObjects(nodeId) ?~! s"Error when deleting system objects (groups, directives, rules) related to relay server '${nodeId.value}'"
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/RemoveNodeService.scala]: this method is missing a logging
it is called by actualNodeDeletion, which returns a Box[JValue], and error is only poped up to the UI or API response

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**